privacyidea/static/components/login/controllers/loginControllers.js

```diff
@@ -102,19 +102,24 @@ angular.module("privacyideaApp")
     $scope.myCountdown = "";
     // We save the previous State in the $rootScope, so that we
     // can return there
-    $rootScope.$on('$stateChangeSuccess',
-        function (ev, to, toParams, from, fromParams) {
-            //debug: console.log("we changed the state from " + from + " to " + to);
-            //debug: console.log(from);
-            //debug: console.log(fromParams);
-            //debug: console.log(to);
+    $transitions.onBefore({},
+        function () {
+            // The stateParams or $state.params are always changed as a reference.
+            // So we need to do a deep copy, to preserve it between state transitions
+            var oldParams = new Array();
```

Comment:
We can use [`angular.copy`](https://docs.angularjs.org/api/ng/function/angular.copy) here

Reply:
This would be nice. Have you checked, if this actually works?
The source is a stateParams object. I am not sure, if it would behave in the same way as the array.

Reply:
done in 66620dda